### PR TITLE
feat: add arm platform support

### PR DESCRIPTION
**Pull request by jessebye**
### Related Issue

https://github.com/codfish/semantic-release-action/issues/206

### Description

Adds ARM64 target to the Docker image build and push.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -62,9 +70,12 @@ jobs:
 
       # Dockerhub is auto synced with the repo, no need to explicitly deploy
       - name: build and push latest docker image to GCR
-        run: |
-          docker build -t ghcr.io/codfish/semantic-release-action:latest .
-          docker push ghcr.io/codfish/semantic-release-action:latest
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          platforms: linux/amd64,linux/amd64
```
Comment (codfish) [inline .github/workflows/release.yml:76]: you mean to have arm here as well?

Comment (jessebye) [reply .github/workflows/release.yml]: d'oh! definitely not - just pushed a commit to use arm64 there.

**Review by codfish — APPROVED** [review]

**Comment by jessebye** [comment]
@codfish for some reason the [release action failed](https://github.com/codfish/semantic-release-action/actions/runs/9530220840/job/26269840791) on the "push docker images for releases to GCR".

The error was: `Error response from daemon: No such image: ghcr.io/codfish/semantic-release-action:latest`, but that doesn't make sense because the previous step shows it pushed that image/tag correctly, and also the [packages page](https://github.com/users/codfish/packages/container/package/semantic-release-action) shows the latest image 🤔 

Any ideas? I'm a bit stumped how it would have pushed it but then failed to tag it immediately after.

**Comment by codfish** [comment]
@jessebye Hey, started to try and combine them yesterday and got sidetracked. I figure they can just be combined now and we can just specify all the tags in that one step you added. Should be able to fix by tomorrow!

My first thought was that maybe it's building in a container or something, out of scope of the runner environment so that future steps can't find the image. Total guess tho but something is definitely going on in that docker action

**Pull request merged**